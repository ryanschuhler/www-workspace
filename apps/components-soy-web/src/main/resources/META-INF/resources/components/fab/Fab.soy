{namespace Fab}

/**
 * FAB icon
 */
{template .render}
  {@param label: string}
  {@param? cssClass: string}
  {@param? handleOnClickFn: any|null}
  {@param? icon: string}
  {@param? id: string}
  {@param? url: string}

  {let $buttonAttributes kind="attributes"}
    aria-label="{$label}"
    class="app-fab--absolute mdc-fab material-icons

    {if $cssClass}
      {sp}{$cssClass}
    {/if}
    "

    data-mdc-auto-init="MDCRipple"

    {if $url}
      href="{$url}"
    {else}
      href="javascript:;"
    {/if}

    {if $id}
      id="{$id}"
    {/if}

    {if $handleOnClickFn}
      onClick="{$handleOnClickFn}"
    {/if}

    title="{$label}"
  {/let}

  <a {$buttonAttributes}>
    <span class="mdc-fab__icon">
      {$icon}
    </span>
  </a>
{/template}