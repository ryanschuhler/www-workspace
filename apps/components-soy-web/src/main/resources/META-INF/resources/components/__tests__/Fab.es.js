'use strict';

import Fab from '../fab/Fab.es';

let component;

describe('Fab', () => {
  afterEach(() => {
    if (component) {
      component.dispose();
    }
  });

  beforeAll(() => {
		window.Liferay = {};
  });

  it('should render', () => {
    component = new Fab({
      icon: 'add',
      label: 'Add Site'
    });

    expect(component).toMatchSnapshot();
  });

  it('should render with url', () => {
    component = new Fab({
      icon: 'add',
      label: 'Add Site',
      url: 'https://www.google.com/'
    });

    expect(component).toMatchSnapshot();
  });
});
