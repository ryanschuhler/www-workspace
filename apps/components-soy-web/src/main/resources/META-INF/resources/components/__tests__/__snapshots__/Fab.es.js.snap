// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fab should render 1`] = `
<a aria-label="Add Site" class="app-fab--absolute mdc-fab material-icons" data-mdc-auto-init="MDCRipple" href="javascript:;" title="Add Site">
  <span class="mdc-fab__icon">add</span>
</a>
`;

exports[`Fab should render with url 1`] = `
<a aria-label="Add Site" class="app-fab--absolute mdc-fab material-icons" data-mdc-auto-init="MDCRipple" href="https://www.google.com/" title="Add Site">
  <span class="mdc-fab__icon">add</span>
</a>
`;
